src/test/storage/chunk_encoder_test.cpp

```diff
@@ -0,0 +1,119 @@
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "all_type_variant.hpp"
+#include "storage/base_value_column.hpp"
+#include "storage/chunk.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/encoded_columns/base_encoded_column.hpp"
+#include "storage/encoded_columns/encoded_columns.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+class ChunkEncoderTest : public BaseTest {
+ public:
+  void SetUp() override {
+    static const auto max_chunk_size = 5u;
+    _table = std::make_shared<Table>(max_chunk_size);
+
+    static const auto column_count = 3u;
+    for (auto column_id = 0u; column_id < column_count; ++column_id) {
+      const auto column_name = std::to_string(column_id);
+      _table->add_column(column_name, DataType::Int);
+    }
+
+    static const auto row_count = max_chunk_size * 3u;
+    for (auto row_id = 0u; row_id < row_count; ++row_id) {
+      const auto row = std::vector<AllTypeVariant>(column_count, AllTypeVariant{static_cast<int32_t>(row_id)});
+      _table->append(row);
+    }
+  }
+
+ protected:
+  void verify_encoding(const Chunk& chunk, const ChunkEncodingSpec& spec) {
+    for (auto column_id = ColumnID{0u}; column_id < chunk.column_count(); ++column_id) {
+      const auto column = chunk.get_column(column_id);
+      const auto column_spec = spec.at(column_id);
+
+      if (column_spec.encoding_type == EncodingType::Invalid) {
```

Comment:
What are cases where this could happen?

Reply:
Okay the usage of `EncodingType::Invalid` as a representation for “unencoded” seems to be confusing. I’ll rename it to `Unencoded`.